src/sentry/types/mypy.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
```

Comment:
can you move this file to `tools/mypy_helpers/plugin.py` ?

- this makes it not part of the `sentry` distributable that goes to prod
- this will allow the current codeowners rules to catch this
- this will ~eventually make it easier to share with `getsentry`